src/test/analyserwaveformtest.cpp

```diff
@@ -8,79 +8,68 @@
 #include "library/dao/analysisdao.h"
 #include "track/track.h"
 
-#define BIGBUF_SIZE (1024 * 1024) //Megabyte
-#define CANARY_SIZE (1024 * 4)
-#define MAGIC_FLOAT 1234.567890f
-#define CANARY_FLOAT 0.0f
-
 namespace {
 
+constexpr std::size_t kBigBufSize = 1024 * 1024; // Megabyte
+constexpr std::size_t kCanarySize = 1024 * 4;
+constexpr float kMagicFloat = 1234.567890f;
+constexpr float kCanaryFloat = 0.0f;
+
 class AnalyzerWaveformTest : public MixxxTest {
   protected:
     AnalyzerWaveformTest()
-            : aw(config(), QSqlDatabase()),
-              bigbuf(nullptr),
-              canaryBigBuf(nullptr) {
+            : m_aw(config(), QSqlDatabase()),
+              m_canaryBigBuf(nullptr) {
     }
 
     void SetUp() override {
-        tio = Track::newTemporary();
-        tio->setAudioProperties(
+        m_pTrack = Track::newTemporary();
+        m_pTrack->setAudioProperties(
                 mixxx::audio::ChannelCount(2),
                 mixxx::audio::SampleRate(44100),
                 mixxx::audio::Bitrate(),
                 mixxx::Duration::fromMillis(1000));
 
-        bigbuf = new CSAMPLE[BIGBUF_SIZE];
-        for (int i = 0; i < BIGBUF_SIZE; i++)
-            bigbuf[i] = MAGIC_FLOAT;
+        // Memory layout for m_canaryBigBuf looks like
+        //   [ canary | big buf | canary ]
 
-        //Memory layout for canaryBigBuf looks like
-        //  [ canary | big buf | canary ]
-
-        canaryBigBuf = new CSAMPLE[BIGBUF_SIZE + 2 * CANARY_SIZE];
-        for (int i = 0; i < CANARY_SIZE; i++)
-            canaryBigBuf[i] = CANARY_FLOAT;
-        for (int i = CANARY_SIZE; i < CANARY_SIZE + BIGBUF_SIZE; i++)
-            canaryBigBuf[i] = MAGIC_FLOAT;
-        for (int i = CANARY_SIZE + BIGBUF_SIZE; i < 2 * CANARY_SIZE + BIGBUF_SIZE; i++)
-            canaryBigBuf[i] = CANARY_FLOAT;
+        m_canaryBigBuf = new CSAMPLE[kBigBufSize + 2 * kCanarySize];
+        for (std::size_t i = 0; i < kCanarySize; i++) {
+            m_canaryBigBuf[i] = kCanaryFloat;
+        }
+        for (std::size_t i = kCanarySize; i < kCanarySize + kBigBufSize; i++) {
+            m_canaryBigBuf[i] = kMagicFloat;
+        }
+        for (std::size_t i = kCanarySize + kBigBufSize; i < 2 * kCanarySize + kBigBufSize; i++) {
+            m_canaryBigBuf[i] = kCanaryFloat;
+        }
```

Comment:
couldn't these be `memset`s?

Reply:
memset() works with char arrays only. We can do a lot more refactoring here, but I like to stop at this point.

Reply:
Right, I forgot about that.